### ISSUE TITLE
Accuracy of Res-18 is lower than expected on ImageNet

### ISSUE THREAD

**Issue by Sunasity** [tensorpack/tensorpack#961]
1. I run the Res-18 by
  + run examples/ResNet/imagenet-resnet.py only with modification on reading data:
    + `def get_imagenet_dataflow(
        datadir, name, batch_size,
        augmentors, parallel=None):
    assert name in ['train', 'val', 'test']
    assert datadir is not None
    assert isinstance(augmentors, list)
    isTrain = name == 'train'
    if parallel is None:
        parallel = min(40, multiprocessing.cpu_count() // 2)  # assuming hyperthreading
    
    lmdb_data = os.path.join(datadir, 'ILSVRC-%s.lmdb'%name)
    ds = LMDBData(lmdb_data, shuffle=False)
    if isTrain:
        ds = LocallyShuffleData(ds, 50000)
    ds = PrefetchData(ds, 5000, 1)
    ds = LMDBDataPoint(ds)
    ds = MapDataComponent(ds, lambda x: cv2.imdecode(x, cv2.IMREAD_COLOR), 0)
    ds = AugmentImageComponent(ds, augmentors, copy=False)
    if parallel < 16:
        logger.warn("DataFlow may become the bottleneck when too few processes are used.")
    ds = BatchData(ds, batch_size, remainder=False)
    ds = PrefetchDataZMQ(ds, parallel)
    ds.reset_state()
        
    return ds`
  + All hyper-parameters are set as default.
2. The top-1 accuracy is 67.5% after 105 epochs, which is 3% lower than reported.
3. Your environment:
  + Python version: 3.6.7
  + TF version: 1.10.0
  + Tensorpack version: 0.8.9
  + Hardware information: 4 * V100

**Comment by Sunasity** [comment]
I am sorry for the wrong editing of the code above. The code for reading data is shown as follows:

```
    lmdb_data = os.path.join(datadir, 'ILSVRC-%s.lmdb'%name)
    ds = LMDBData(lmdb_data, shuffle=False)
    if isTrain:
        ds = LocallyShuffleData(ds, 50000)
    ds = PrefetchData(ds, 5000, 1)
    ds = LMDBDataPoint(ds)
    ds = MapDataComponent(ds, lambda x: cv2.imdecode(x, cv2.IMREAD_COLOR), 0)
    ds = AugmentImageComponent(ds, augmentors, copy=False)
    if parallel < 16:
        logger.warn("DataFlow may become the bottleneck when too few processes are used.")
    ds = BatchData(ds, batch_size, remainder=False)
    ds = PrefetchDataZMQ(ds, parallel)
    ds.reset_state()
```

**Comment by ppwwyyxx** [comment]
The original models were trained without modifications to the code and on 8 GPUs. Since you've made changes to the code, getting different results is not surprising.

**Comment by Sunasity** [comment]
> The original models were trained without modifications to the code and on 8 GPUs. Since you've made changes to the code, getting different results is not surprising.

Hi, I conduct the experiments with the original code on 8 GPUs. The top-1 accuracy is still 67.5% (3% lower than reported).

**Comment by ppwwyyxx** [comment]
Here is the log 4 days ago that reproduced the experiments, with no modification to the code.
[log.log](https://github.com/tensorpack/tensorpack/files/2551563/log.log)

I'm using TF 1.11, cuda 9.0.176, cudnn 7.2.1

**Comment by ppwwyyxx** [comment]
Could you provide your environment info (especially TF version and cudnn version)?

**Comment by Sunasity** [comment]
> Could you provide your environment info (especially TF version and cudnn version)?

My TF version is 1.10.0 and the cudnn version is 9.0.

**Comment by ppwwyyxx** [comment]
> cudnn version is 9.0

9.0 is cuda version.

Made another run with TF 1.10, cuda 9.0, cudnn 7.1.1 and still able to reproduce results.
I recommend you to take a look at https://docs.nvidia.com/deeplearning/sdk/cudnn-release-notes/index.html which does mention some bugs in certain cudnn versions.

**Comment by ppwwyyxx** [comment]
Made another run with TF1.12, cuda9.0, cudnn7.4.1, latest tensorpack mater on Jan 14, unmodified example code. Observed the same results. Closing as the issue cannot be reproduced and is likely an environment issue.
